### PR TITLE
Fix a bug in old gen policy

### PR DESCRIPTION
**Pull request by rguo-aws**
*Fixes #:*

*Description of changes:*
found a bug in end to end dev testing. The bucketization function had a bug and gave wrong bucket index. Fixing the bug and updating the corresponding UT to capture it. 

*Tests:*
tested on dev stack 

*If new tests are added, how long do the new ones take to complete*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pa_config/rca.conf [hunk 1]
```diff
@@ -93,46 +93,5 @@
     "ClusterTemperatureRca"
   ],
   "muted-deciders": [],
-  "muted-actions": [],
```
Comment (vigyasharma) [inline pa_config/rca.conf:96]: We need to clean up these configs, `muted_actions` and `muted_deciders` are also only needed by master node. Can you create an issue for this?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/jvm/old_gen/LevelTwoActionBuilder.java [hunk 2]
```diff
@@ -150,7 +150,7 @@ private void actionPriorityForCache() {
    */
   private int bucketization(int lowerBound, int upperBound, int val, int bucketSize) {
     double step = (double) (upperBound - lowerBound) / (double) bucketSize;
-    return (int) ((double) val / step);
+    return (int) ((double) (val - lowerBound) / step);
```
Comment (vigyasharma) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/jvm/old_gen/LevelTwoActionBuilder.java:153]: nit: can we rename value by `queueSize` or `size` or `resourceSize` ?

File: pa_config/rca_master.conf [hunk 3]
```diff
@@ -109,20 +109,47 @@
   // Default Values are added in the config specific file as well
   // Modify them as well whenever changing the default values (DeciderConfig.java)
   "decider-config-settings": {
-    // Decreasing order of priority for the type of workload we can expect on the cluster.
-    // Priority order in the list goes from most expected to the lease expected workload type.
     "workload-type": {
-      "priority-order": ["ingest", "search"]
+      "prefer-ingest": true,
+      "prefer-search": true
```
Comment (sidheart) [inline pa_config/rca_master.conf:114]: The other format seems much clearer to me. Why did we change it to this? What does this mean?

**Review by vigyasharma — APPROVED** [review]

File: pa_config/rca_master.conf (hunk 3 above)
Comment (rguo-aws) [reply pa_config/rca_master.conf:114]: Specifying the priority order as an array has two issues : 
1. we can't assign search and index operations with the same level of priority
2. we need maintain a function to parse all elements in array

**Review by sidheart — APPROVED** [review]

**Pull request merged**